data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/schema/BeaconValidatorsResponse.java

```diff
@@ -18,8 +18,43 @@
 
 public class BeaconValidatorsResponse {
   public final SSZList<Validator> validatorList;
+  private int totalSize;
+  private int nextPageToken;
 
   public BeaconValidatorsResponse(SSZList<Validator> validatorList) {
-    this.validatorList = validatorList;
+    this(validatorList, 20, 0);
+  }
+
+  public BeaconValidatorsResponse(
+      SSZList<Validator> validatorList, final int pageSize, final int pageToken) {
+    // first page is pageToken = 0
+    if (pageSize > 0 && pageToken >= 0) {
+      int offset = pageToken * pageSize;
+      SSZList<Validator> pageOfValidators = new SSZList<>(Validator.class, pageSize);
```

Comment:
i'm not sure in the response we want to be using SSZList unless we've got a good reason. we can just use ArrayList and not have the baggage... they serialize the same way
We might find this logic to be a useful utility function, passing in a list (I'd probably make it a List, as both SSZVector and SSZList come from Arraylist from memory) and getting a page of data...

since it's a list, we could possibly just use list.sublist, and pass it the start and end and return the result of that... also can just return List.of() for empty...

Reply:
as per discussion, can't simply use List.sublist because of the indexing. Other suggestions taken on board. SSZList is gone from Response and now using List.of() for empty